### PR TITLE
FC-1438: fix/predicate dash

### PR DESCRIPTION
**Pull request by zonotope**
The fix is in fad3b1535275c073c35eb391813e1aab21a046b8, and the rest is a little cleanup. 

We were using two different parsing/validation criteria for transacting and querying, with the transaction one being more permissive. I couldn't find a reference to any (dis)allowed characters in predicate names in our docs so I decided to go with the most permissive criteria possible on the querying side, leaving the transact validations as they were.

### CODE REVIEW

**Comment by zonotope** [comment]
@fluree/core I see a chance for some more refactoring on the ledger side that I'd like to explore, so I'm coverting this pr to draft for now.

**Comment by zonotope** [comment]
@fluree/core the thing I wanted to try was more trouble than it's worth. this is ready for review again.

File: src/fluree/db/query/analytical.cljc [hunk 1]
```diff
@@ -40,28 +40,34 @@
...
-                            (not (dbproto/-p-prop db :name (re-find #"[_a-zA-Z0-9/]*" key))))
+                 (when (and (= idx 1)
+                            (not key-as-var)
+                            (not= "_id" key)
+                            (not (dbproto/-p-prop db :name (parse/as-predicate-name key))))
                    (throw (ex-info (str "Invalid predicate provided: " key)
-                                   {:status 400
-                                    :error  :db/invalid-query})))
+                                                 {:status 400
+                                                  :error  :db/invalid-query})))
```
Comment (cap10morgan) [inline src/fluree/db/query/analytical.cljc:70]: undo indentation?

**Review by cap10morgan — APPROVED** [review]
One small formatting thing but LGTM otherwise!

**Pull request merged**